### PR TITLE
Game logic: Milestone 3

### PR DESCRIPTION
**Pull request by oloomoses**
In this milestone:
* We created a feature branch called the game_logic
* We added two classes; Players and Board class in the lib folder
* We created the game logic which enables two users to play the game
* The game shows whether its a win or a draw and displays the winner

### CODE REVIEW

File: lib/board.rb
Comment (euqueme) [inline lib/board.rb]: - [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- Refactor the display method so instead of being in charge of printing the board is in charge only to retrieve the board in a string that can be printed in the `main.rb` file doing something like: `puts play_ground.display(board_data)`

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @oloomoses and @gabrie-lhilarion :lollipop: :balloon: 

### Thank you for your submission :handshake: your project works very well and you've used OOP practices brilliantly :sparkles: I personally like the fact that you request for the player's names this makes your game stand out from the rest :video_game: :1st_place_medal: let's work on the following changes :construction: :construction_worker_man: :construction_worker_man: 

#### General
- [ ] Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

- You have both rubocop and stickler files in your project but it doesn't seem to be working, please activate the repo in stickler and don't submit until your last commit has a green checkmark :heavy_check_mark: 

#### Behaviour

- [ ] Your project is letting the players have empty names, let's add a validation to prevent the players from having empty names or provide one when no name is given

image

#### Milestone 3 specifications

- [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- I've left a comment directly in your code to explain where you need to do the change

### That's it, team! :raised_hands: 

### Please :pray:  do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Great work, congratulation :+1: Milestone Approved.
You did all changes required previously. 
I would suggest two more optional changes before you merge this Pull Request:
* You can delete the `.gitkeep` file as it is no longer needed (it is used only for committing empty directories).
* Game will look better if you add an empty line between the board and a message about players' moves. 

Then go ahead and merge to development.

_Goodluck in your next milestone._ *Happy coding*

**Pull request merged**
